core/src/ast/data.rs

```diff
@@ -150,17 +152,48 @@ impl<V: UsesLifetimes, F: UsesLifetimes> UsesLifetimes for Data<V, F> {
 }
 
 /// Equivalent to `syn::Fields`, but replaces the AST element with a generic.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone)]
 pub struct Fields<T> {
     pub style: Style,
     pub fields: Vec<T>,
+    // This field is intentionally private, to make it easier to add new fields to this struct
```

Comment:
Rather than making this field private, how about we add it as public and add `__nonexhaustive: ()` as a second field? That has no runtime impact, but lets people continue to use the field as public.

Reply:
I do not know if this would be good. Most people expect to obtain a `Span` with `Spanned::span`, but in this case the compiler would suggest to use the field instead.

Reply:
Oh, that's a really good point.